File: Serie/src/main/java/com/ferrazeduardo/escola/Serie/domain/entities/Professor.java
package com.ferrazeduardo.escola.Serie.domain.entities;

import java.util.List;
import java.util.UUID;

public class Professor {
    public Professor(UUID id, String nome, List<Materia> materia) {
        Id = id;
        this.nome = nome;
        this.materia = materia;
    }

    private UUID Id;

    public UUID getId() {
        return Id;
    }

    public String getNome() {
        return nome;
    }

    public List<Materia> getMateria() {
        return materia;
    }

    private String nome;
    private List<Materia> materia;

}

File: Serie/Serie.Domain/Entity/Professor.cs
using System;

namespace Serie.Domain.Entity;

public class Professor : SeedWorks.Entity
{
    public Professor(Guid id, String nome, ICollection<Materia> materia)
    {
        this.setId(id);
        this.NM_PROFESSOR = nome;
        this.Materias = materia;
    }

    public String getNome()
    {
        return NM_PROFESSOR;
    }

    public ICollection<Materia> getMateria()
    {
        return Materias;
    }

    public String NM_PROFESSOR { get; private set; }
    public ICollection<Materia> Materias { get; private set; } = [];

}
